Formula/gtk4.rb

```diff
@@ -33,18 +33,24 @@ class Gtk4 < Formula
 
   uses_from_macos "libxslt" => :build # for xsltproc
 
+  on_linux do
+    depends_on "libxkbcommon"
+  end
+
   def install
     args = std_meson_args + %w[
-      -Dx11-backend=false
-      -Dmacos-backend=true
       -Dgtk_doc=false
       -Dman-pages=true
       -Dintrospection=enabled
       -Dbuild-examples=false
       -Dbuild-tests=false
     ]
 
-    args << "-Dprint-cups=disabled" if MacOS.version <= :mojave
+    on_macos do
+      args << "-Dx11-backend=false"
+      args << "-Dmacos-backend=true"
+      args << "-Dprint-cups=disabled" if MacOS.version <= :mojave
```

Comment:
Will this enable `cups` on Linux? If so, then wouldn't you need a `uses_from_macos "cups"`? Or is this there to fix broken `cups` autodetection on macOS?

Reply:
Note that `MacOS.version <= :mojave` will evaluate to `true` on Linux.

Reply:
This line was already existing, I just moved it inside the `on_macos` block since it had the macos version check. Since I was just trying to get the Linux build to work, I didn't look at the build output on macos to see what it does.

Reply:
Yes, but moving it into the `on_macos` block means that you no longer add `-Dprint-cups=disabled` on Linux, and this clearly affects the Linux build. Is that something you want to do on Linux?

Reply:
ah ok, I will try another build on Linux

Reply:
It builds without error (cups was not detected)

Reply:
What about if you happen to already have `cups` installed? Usually we disable stuff to prevent opportunistic linkage when users try to build when they are installed.

Reply:
I imagine that it would detect it and enable it. So perhaps strictly enable it on Linux?

Reply:
That's what
```ruby
args << "-Dprint-cups=disabled" if MacOS.version <= :mojave
```
does when you leave it outside of the `on_macos` block.

Reply:
I added "cups" as a dependency on Linux so that it will always be enabled.

Reply:
> ```ruby
> args << "-Dprint-cups=disabled" if MacOS.version <= :mojave
> ```

Wouldn't this disable it rather than enable it? I went for the option of enabling it.

Reply:
Ah, yes, I had misread your last message.